Data/Primitive/UnliftedArray.hs

```diff
@@ -137,6 +141,22 @@ instance PrimUnlifted (MV.MutVar s a) where
   toArrayArray# (MV.MutVar mv#) = unsafeCoerce# mv#
   fromArrayArray# aa# = MV.MutVar (unsafeCoerce# aa#)
 
+instance PrimUnlifted (MVar a) where
+  toArrayArray# (MVar mv#) = unsafeCoerce# mv#
```

Comment:
@dolio  Yes, and i extend your trick to many more unlifted boxed type wrapper, such as `IORef/STRef` and `MVar/TVar`, i think @cartazio didn't get this at first, maybe your explain will make him feel ensured.